lib/parsers.js

```diff
@@ -15,7 +15,7 @@ module.exports.parseText = function parseText(message, kwargs) {
 
 module.exports.parseError = function parseError(err, kwargs, cb) {
   utils.parseStack(err, function(frames) {
-    var name = err.name + '';
+    var name = err.constructor.name + '';
```

Comment:
Should it not be:

```javascript
var name = (err.name || err.constructor.name) + '';
```

That way if someone sets a local name property, that is prioritized over going up the constructor chain?

Reply:
If one won't set `err.name` manually, then it defaults to `Error`, which will result in `err.constructor.name` to be never used.

Reply:
How about:

```javascript
var name = {}.hasOwnProperty.call(err, 'name') ?
  err.name : err.constructor.name;
```

Reply:
Got it. Updated and added additional tests covering this cases.